app/screens/home/search/results/results.tsx

```diff
@@ -186,9 +198,18 @@ const SearchResults = ({
         const container: StyleProp<ViewStyle> = fileInfos.length > 1 ? styles.container : undefined;
         const isSingleImage = orderedFilesForGallery.length === 1 && (isImage(orderedFilesForGallery[0]) || isVideo(orderedFilesForGallery[0]));
         const isReplyPost = false;
+        const optionSelected = (selectedItemNumber !== undefined) && (selectedItemNumber === filesForGalleryIndexes[item.id!]);
```

Comment:
This part (the component of each file) starts to be really complex. 0/5 on whether it makes sense to move this to its own component.

Reply:
I did a significant refactor to help reduce the logic for each component.  Please see the PR description notes regarding the refactor.